src/lock/optiql.cpp

```diff
@@ -168,7 +184,8 @@ OptiQL::GetVersion()  //
   SpinWithBackoff(
       [](const std::atomic_uint64_t *lock, uint64_t *cur) -> bool {
         *cur = lock->load(kAcquire);
-        return (*cur & kXLock) == kNoLocks;
+        return (*cur & kXLock) == kNoLocks  // lock released
+               || (*cur & kOPReadLock);     // or, opportunistically readable
```

Comment:
```cpp
constexpr uint64_t kXAndOPReadMask = kXLock | kOPReadLock;
```

みたいなのを作って，

```suggestion
        return (*cur & kXAndOPReadMask) != kXLock;
```

とした方がシンプルでよい気がします．